src/components/Filters/SchemaSieve.ts

```diff
@@ -112,21 +111,15 @@ export const createFullTextSearchFilter = (
 	return filter as JSONSchema;
 };
 
-// Update or insert a full text search filter into an array of filters
-export const upsertFullTextSearch = (
+// Insert a full text search filter into an array of filters
+export const insertFullTextSearch = (
 	schema: JSONSchema,
 	filters: JSONSchema[] = [],
 	term: string,
 ) => {
 	const searchFilter = createFullTextSearchFilter(schema, term);
-	const existingSearchIndex = findIndex(filters, { title: FULL_TEXT_SLUG });
 
-	if (existingSearchIndex > -1) {
-		return filters.map((filter, i) =>
-			existingSearchIndex === i ? searchFilter : filter,
-		);
-	}
```

Comment:
By removing this piece, we will no longer be able to override a search made with the search bar as they will be added in &. This removes the override behaviour of the previous filter, is this something we really want?

Reply:
Well the overriding behavior of the previous filter is one of the things that created the illusion that the search bar is a *focus search*. With these changes, we will no longer replace the old filter but rather add a new one when the user hits enter (classic search behavior). If the user does not hit enter, they are searching through their filtered list and getting a preview of their search results in the focus search dropdown.

If what you are saying is that hitting enter should replace the old entry you made by hitting enter, I believe this was a point we considered in FD. See the following thread: https://www.flowdock.com/app/rulemotion/resin-frontend/threads/A2X-xEh5uPoGVTR-h7a8p3pb71X
Me: "Should pressing enter to add the term to the filter replace the current fulltext filter? Or should it add another filter? (for example I press enter on 'rf'. If I press enter on 'l', in its current state the 'l' filter replaces the 'rf' filter, but I'm thinking it should now have both filters instead. Not 100% sure which is why I'm checking)"
Chris: "I think it should add a second filter; each press of enter should keep adding filters"

Reply:
I'm lost when you say `classic search behavior`, I can't think of any site that currently does this type of search with a search bar (adding N filters in AND). Could you give me an example ?

<img width="1623" alt="Screenshot 2021-11-15 at 16 33 45" src="https://user-images.githubusercontent.com/7238159/141809919-853dced9-b033-4823-920f-7ff3376b83c3.png">

IMHO something like this can be really confusing

Reply:
I hear your argument. I could go either way I suppose. I saw it as adding a fulltext filter rather than submitting a *new* search. Perhaps that's the wrong way to look at it. What do you think @chrisys ?

Reply:
@JSReds If you want I can revert to upsert for this PR and then we can switch to insert in a separate followup PR if we decide that's really the functionality we want

Reply:
Sounds good to me 👍

Reply:
Readding insert as per our meeting discussion. Will determine whether to switch back to upsert based on user feedback